### PR TITLE
sortwithwswaps

### PR DESCRIPTION
**Pull request by mengyuaw**

### CODE REVIEW

File: SortWithSwapsPlugin.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+public class SortWithSwapsPlugin implements GamePlugin {
+    private int ONE = 1;
+    private int TEN = 10;
+
+    // Describes whether the current move is over
+    public Boolean move = false;
+
+    // An array of the numbers 1-10
+    private int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
```
Comment (Doris-Ge) [inline SortWithSwapsPlugin.java:16]: arr should be final

File: SortWithSwapsPlugin.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    // The game framework
+    private GameFramework framework = null;
```
Comment (Doris-Ge) [inline SortWithSwapsPlugin.java:19]: no need to initialize to null

File: SortWithSwapsPlugin.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    // The last number clicked, which is to be swapped with the next number clicked
+    private int last = -1;
+
+    //The number of swaps the user has made
+    static Integer count = null;
```
Comment (Doris-Ge) [inline SortWithSwapsPlugin.java]: use int and make it private

File: SortWithSwapsPlugin.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    public Boolean move = false;
+
+    // An array of the numbers 1-10
+    private int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+
+    // The game framework
+    private GameFramework framework = null;
+
+    // The last number clicked, which is to be swapped with the next number clicked
+    private int last = -1;
```
Comment (Doris-Ge) [inline SortWithSwapsPlugin.java:22]: make it final

**Review by Alvinlst — APPROVED** [review]

**Pull request merged**